gmso/formats/mol2.py

```diff
@@ -0,0 +1,219 @@
+"""Convert to and from a TRIPOS mol2 file."""
+# TODO add sources of mol2 files
+import warnings
+from pathlib import Path
+
+import unyt as u
+
+from gmso import Atom, Bond, Box, Topology
+from gmso.core.element import element_by_name, element_by_symbol
+from gmso.formats.formats_registry import loads_as
+
+
+@loads_as(".mol2")
+def from_mol2(filename, site_type="atom"):
+    """Read in a TRIPOS mol2 file format into a gmso topology object.
+
+    Creates a Topology from a mol2 file structure. This will read in the
+    topological structure (sites, bonds, and box) information into gmso.
+    Note that parameterized information can be found in these objects, but
+    will not be converted to the Topology.
+
+    Parameters
+    ----------
+    filename : string
+        path to the file where the mol2 file is stored.
+    site_type : string ('atom' or 'lj'), default='atom'
+        tells the reader to consider the elements saved in the mol2 file, and
+        if the type is 'lj', to not try to identify the element of the site,
+        instead saving the site name.
+
+    Returns
+    -------
+    top : gmso.Topology
+
+    Notes
+    -----
+    It may be common to want to create an mBuild compound from a mol2 file. This is possible
+    by installing [mBuild](https://mbuild.mosdef.org/en/stable/index.html)
+    and converting using the following python code:
+
+        >>> import gmso.Topology
+        >>> from gmso.external.convert_mbuild import to_mbuild
+        >>> top = Topology.load('myfile.mol2')
+        >>> mbuild_compound = to_mbuild(top)
+    """
+    path = Path(filename)
+    if not path.exists():
+        msg = "Provided path to file that does not exist"
+        raise OSError(msg)
```

Comment:
A better/fitting exception should be [`FileNotFoundError`](https://docs.python.org/3/library/exceptions.html#FileNotFoundError)
```suggestion
        raise FileNotFoundError(msg)
```